lib/internal/fs/rimraf.js

```diff
@@ -208,10 +209,12 @@ function _rmdirSync(path, options, originalErr) {
         rimrafSync(join(path, child), options);
       });
 
-      for (let i = 0; i < options.maxRetries + 1; i++) {
+      for (let i = 1; i <= options.maxRetries + 1; i++) {
         try {
           return rmdirSync(path, options);
-        } catch {} // Ignore errors.
+        } catch {
+          sleep(i * options.retryDelay);
```

Comment:
`options.retryDelay` can be undefined:

https://github.com/nodejs/node/blob/bd9be0418a4ce528c3497ff80d5d70404be10f6f/lib/internal/fs/utils.js#L588

Note how it passes `0` as the default but discards the return value (and also allows values < 0? wuh?)

That means this statement can evaluate to `sleep(-42)`, or `sleep(i * undefined)` / `sleep(Nan)`, which hits the `validateUint32()` check in `sleep()`.

Tangential observation: the coercion-to-NaN works out for the `i <= options.maxRetries + 1` check above because `i <= NaN + 1` is always false.

Reply:
`validateInt32()` doesn't have a return value. It just throws if the input isn't an "int32" in JS. The 0 is the minimum allowed value.

Reply:
Oh, you're right... I got confused by `parseMode()` and `validateRmdirOptions()`, which do return values.

Okay, objection withdrawn but then the logic should perhaps be:
```js
const { retryDelay } = options;
if (retryDelay > 0) sleep(i * retryDelay);  // or handle that in sleep()
```
Because `sleep(0)` will still yield the process (at least on Linux and probably other Unices too.)

Reply:
OK, updated to prevent sleeping for delays of 0.